Code/src/main/java/nl/utwente/group10/ui/components/blocks/Block.java

```diff
@@ -93,8 +97,46 @@ public final CustomUIPane getPane() {
      * change.
      *
      * This method should only be called after the Block's constructor is done.
+     * 
+     * This method will invalidate the Block even if the state did not change.
+     */
+    public void invalidateConnectionState() {
+    }
+
+    /**
+     * Does the same as invalidateConnectionState(), but cascading down to
+     * other blocks which are possibly also (indirectly) affected by the state
```

Comment:
Weet je heul heul *heul* zeker dat alleen blokken *downstream* geinvalidate hoeven te worden? Het zou best kunnen, maar het is denk ik niet triviaal.